packages/sampling/src/sampler.ts

```diff
@@ -411,9 +416,17 @@ export abstract class OboeSampler implements Sampler {
   abstract toString(): string
 
   #getSettings(...params: SampleParams): Settings | undefined {
-    return (
-      this.#settings && merge(this.#settings, this.localSettings(...params))
-    )
+    if (!this.#settings) {
+      return
+    }
+
+    const expiry = this.#updated + this.#settings.ttl * 1000
+    if (Date.now() > expiry) {
+      this.logger.debug("settings expired")
```

Comment:
Is it better to remove the expired setting here?

Reply:
Good idea